internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -1182,13 +1191,10 @@ func addOptionalOauthClientFields(ctx context.Context, addRequest *client.Client
 		addRequest.PersistentGrantReuseGrantTypes = slice
 	}
 
-	if internaltypes.IsDefined(plan.RestrictScopes) {
-		addRequest.RestrictScopes = plan.RestrictScopes.ValueBoolPointer()
-		if *plan.RestrictScopes.ValueBoolPointer() && internaltypes.IsDefined(plan.RestrictedScopes) {
-			var slice []string
-			plan.RestrictedScopes.ElementsAs(ctx, &slice, false)
-			addRequest.RestrictedScopes = slice
-		}
+	if *plan.RestrictScopes.ValueBoolPointer() && internaltypes.IsDefined(plan.RestrictedScopes) {
+		var slice []string
+		plan.RestrictedScopes.ElementsAs(ctx, &slice, false)
```

Comment:
Would it make more sense for this if logic to just be part of validate config instead? It seems like silently ignoring the restricted_scopes value rather than printing an explicit error is harder for a user to understand.